src/basic/socket-util.c

```diff
@@ -345,6 +345,25 @@ int sockaddr_port(const struct sockaddr *_sa, unsigned *ret_port) {
         }
 }
 
+union in_addr_union *sockaddr_in_addr(const struct sockaddr *_sa) {
```

Comment:
No, it's like strstr(): accept const, return non-const, even though it's the same object we return a pointer to. That's because this means we can sanely use it both on const and on non const arguments, and use the result both in const and non-const contexts. if we'd return "const" here, then it would still accept const + non-const arguments, but even if you pass a non-const (i.e. writable) structure the return value would be const (i.e. read-only), wich is unnecessarily limiting.